### DISCUSSION TITLE
Spatie Laravel Media Library Plugin Field in Builder Field

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3989] [Help]
Is it possible to use the `SpatieMediaLibraryFileUpload` field as part of a Builder field? I'm able to save the resource with two blocks and it looks like it works - but when I reload the page, both blocks show the same image.

**Comment by danharrin** [comment]
No, that doesn't work unfortunately. Spatie associates files with models, not builder JSON keys.

**Reply by intrepidws** [reply]
Is it possible to use Builder backed by models, rather than JSON? Perhaps a Blocks model, or something like that?

**Reply by danharrin** [reply]
Yeah that's an idea, havent had time to explore it though

**Reply by intrepidws** [reply]
Another idea that I had is to actually store the media model ID in the JSON for Builder/Repeater fields. That would be a way for the system to know which media goes with which block. I looked into implementing something like that, but unfortunately I'm not nearly that smart :)
